Automattic/jetpack
# Blocks: convert to module

Fixes https://github.com/Automattic/jetpack/issues/39351
Fixes #39441
Supercedes and closes #39422

## Proposed changes:

* Drop use of an option for tracking if blocks should be used; deferring to the existing module system.

This will reduce the db calls, since this option is typically not set until someone has toggled it the first time, and convert it to a typical entity.

### Other information:


## Jetpack product discussion


n/a

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:

A few different things to test. Each should be tested on a clean site (e.g. Nothing Jetpack on it; either new sites or uninstall Jetpack / delete jetpack_options, etc.)

Default new install state (auto-activation of the blocks module)
* Apply this patch.
* Setup a regular Jetpack site, site-only connection at least.
* Verify that the blocks module is active ( /wp-admin/admin.php?page=jetpack_modules )
* Verify that Jetpack->Settings->Writing has the Blocks option toggled on.
* Verify that the post block editor has the Jetpack blocks.

Existing site (auto-activation of a newly defined module)
* On a stable Jetpack, site-only connection.
* Verify there is no blocks module ( /wp-admin/admin.php?page=jetpack_modules )
* Using the beta plugin, update Jetpack to this branch.
* Check the same as above to ensure the module is activated on upgrade.

Existing site with the option previously set true (confirms the module does not activate to respect previous setting and the old option has been deleted to clean up after ourselves.
* On a stable Jetpack, site-only connection at least.
* Either directly set the option (e.g. `wp option set jetpack_blocks_disabled true`) or visit Jetpack->Settings->Writing and toggle the block option.
* Go to the post editor to confirm blocks are not loading.
* Update to this branch.
* Verify that the blocks module is NOT active ( /wp-admin/admin.php?page=jetpack_modules )
* Verify that Jetpack->Settings->Writing has the Blocks option toggled OFF.
* Verify that the post block editor DOES NOT have the Jetpack blocks.
* Verify that the previous option is _still_ present.
* Activate the blocks module.
* Verify the previous option was deleted `wp option get jetpack_blocks_disabled` should return `Error: Could not get 'jetpack_blocks_disabled' option. Does it exist?`


Existing site with the option previously set false. (if someone had previously disabled and re-enabled; verifies that the modules are active to match previous setting and option has been deleted to clean up)
* On a stable Jetpack, site-only connection at least.
* Either directly set the option (e.g. `wp option set jetpack_blocks_disabled false`) or visit Jetpack->Settings->Writing and toggle the block option to disable and then re-enable the blocks.
* Go to the post editor to confirm blocks are loading.
* Update to this branch.
* Verify that the blocks module is active ( /wp-admin/admin.php?page=jetpack_modules )
* Verify that Jetpack->Settings->Writing has the Blocks option toggled on.
* Verify that the post block editor has the Jetpack blocks.
* Verify the previous option was deleted `wp option get jetpack_blocks_disabled` should return `Error: Could not get 'jetpack_blocks_disabled' option. Does it exist?`
